crates/typst/src/eval/plugin.rs

```diff
@@ -0,0 +1,204 @@
+use crate::diag::SourceResult;
+use crate::util::Bytes;
+use ecow::EcoString;
+use std::sync::Mutex;
+use typst::diag::At;
+use wasmi::{AsContext, Caller, Engine, Func as Function, Linker, Module, Value};
+
+#[derive(Debug)]
+pub struct Plugin(Mutex<PluginInstance>, Bytes);
+
+impl crate::eval::Type for Plugin {
+    const TYPE_NAME: &'static str = "Plugin";
+}
+
+impl PartialEq for Plugin {
+    fn eq(&self, other: &Self) -> bool {
+        self.1 == other.1
+    }
+}
+
+impl std::hash::Hash for Plugin {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.1.hash(state);
+    }
+}
+
+impl Plugin {
+    pub fn new(instance: PluginInstance, bytes: Bytes) -> Self {
+        Self(Mutex::new(instance), bytes)
+    }
+    pub fn call(
+        &self,
+        function: &str,
+        args: &mut typst::eval::Args,
```

Comment:
You can import some more types. Here, `use super::Args` because we are already in eval. Also, within `typst`, using `crate` is preferrable over `typst`.